### PR TITLE
Simplify example training script and agent

### PR DESCRIPTION
**Pull request by glmcdona**
Simplify reward, metrics, and logic.

* Reward is now just a delta reward per unit, citytile, and resources collected. Also a reward for each city tile end of game.
* Removed stats logging.
* Removed more smart city research logic.
* Tuned some comments.
* Removed train script game-details tensorflow custom logger.

### CODE REVIEW

**Review by nosound2 — APPROVED** [review]
Thanks, looks good to me
